wallet/wallet-controller/src/lib.rs

```diff
@@ -1138,12 +1144,10 @@ where
                 .iter()
                 .enumerate()
                 .map(|(i, txo)| {
-                    let htlc_spending =
-                        htlc_secrets.as_ref().map_or(HtlcSpendingCondition::Skip, |secrets| {
-                            secrets.get(i).map_or(HtlcSpendingCondition::WithMultisig, |_| {
-                                HtlcSpendingCondition::WithSecret
-                            })
-                        });
+                    let htlc_spending = HtlcSpendingCondition::from_opt_secrets_array_item(
+                        htlc_secrets.as_deref(),
+                        i,
+                    );
 
```

Comment:
FYI: this is actually a fix and not just cleanup - in the old code the `_` parameter of the inner closure would have the type `&Option<HtlcSecret>` (due to `secrets.get` returning `Option` of `Option`), so we would always use `HtlcSpendingCondition::WithSecret` when Some(Vec) was passed here for `htlc_secrets`.